<html><body><div><div role="main">
        
<h2>Security releases issued: 1.8.3, 1.7.9, 1.4.21</h2>

<span class="meta">Posted by <strong>Tim Graham</strong> on July 8, 2015 </span>

<p>In accordance with <a class="reference external" href="https://docs.djangoproject.com/en/dev/internals/security/">our security release policy</a>, the Django team is issuing multiple releases -- Django 1.4.21, 1.7.9, and 1.8.3. These releases are now available on PyPI and our download page. These releases address several security issues detailed below. We encourage all users of Django to upgrade as soon as possible. The Django master branch has also been updated.</p>
<div class="section" id="s-denial-of-service-possibility-by-filling-session-store">
<h3>Denial-of-service possibility by filling session store</h3>
<p>In previous versions of Django, the session backends created a new empty record
in the session storage anytime <tt class="docutils literal">request.session</tt> was accessed and
there was a session key provided in the request cookies that didn't
already have a session record. This could allow an attacker to easily
create many new session records simply by sending repeated requests
with unknown session keys, potentially filling up the session store or
causing other users' session records to be evicted.</p>
<p>The built-in session backends now create a session record only if the
session is actually modified; empty session records are not created.
Thus this potential DoS is now only possible if the site chooses to
expose a session-modifying view to anonymous users.</p>
<p>As each built-in session backend was fixed separately (rather than a
fix in the core sessions framework), maintainers of third-party
session backends should check whether the same vulnerability is
present in their backend and correct it if so.</p>
<p>Thanks Eric Peterson and Lin Hua Cheng for reporting the issue.</p>
<p>This issue has been assigned the identifier CVE-2015-5143.</p>
</div>
<div class="section" id="s-header-injection-possibility-since-validators-accept-newlines-in-input">
<h3>Header injection possibility since validators accept newlines in input</h3>
<p>Some of Django's built-in validators
(<tt class="docutils literal">django.core.validators.EmailValidator</tt>, most seriously) didn't
prohibit newline characters (due to the usage of <tt class="docutils literal">$</tt> instead of <tt class="docutils literal">\Z</tt> in the
regular expressions). If you use values with newlines in HTTP response or email
headers, you can suffer from header injection attacks. Django itself isn't
vulnerable because <tt class="docutils literal">django.http.HttpResponse</tt> and the mail sending
utilities in <tt class="docutils literal">django.core.mail</tt> prohibit newlines in HTTP and SMTP
headers, respectively. While the validators have been fixed in Django, if
you're creating HTTP responses or email messages in other ways, it's a good
idea to ensure that those methods prohibit newlines as well. You might also
want to validate that any existing data in your application doesn't contain
unexpected newlines.</p>
<p><tt class="docutils literal">django.core.validators.validate_ipv4_address()</tt>,
<tt class="docutils literal">django.core.validators.validate_slug()</tt>, and
<tt class="docutils literal">django.core.validators.URLValidator</tt> are also affected, however, as
of Django 1.6 the <tt class="docutils literal">GenericIPAddresseField</tt>, <tt class="docutils literal">IPAddressField</tt>, <tt class="docutils literal">SlugField</tt>,
and <tt class="docutils literal">URLField</tt> form fields which use these validators all strip the input, so
the possibility of newlines entering your data only exists if you are using
these validators outside of the form fields.</p>
<p>The undocumented, internally unused <tt class="docutils literal">validate_integer()</tt> function is now
stricter as it validates using a regular expression instead of simply casting
the value using <tt class="docutils literal">int()</tt> and checking if an exception was raised.</p>
<p>Thanks Sjoerd Job Postmus for reporting the issue.</p>
<p>This issue has been assigned the identifier CVE-2015-5144.</p>
</div>
<div class="section" id="s-denial-of-service-possibility-in-url-validation">
<h3>Denial-of-service possibility in URL validation</h3>
<p><tt class="docutils literal">django.core.validators.URLValidator</tt> included a regular expression
that was extremely slow to evaluate against certain inputs. This regular
expression has been simplified and optimized.</p>
<p>Thanks Jo√£o Silva and Ross Brunton for reporting the issue.</p>
<p>This issue has been assigned the identifier CVE-2015-5145.</p>
</div>
<div class="section" id="s-affected-supported-versions">
<h3>Affected supported versions</h3>
<ul class="simple">
<li>Django master development branch</li>
<li>Django 1.8</li>
<li>Django 1.7 (except the URL DoS issue)</li>
<li>Django 1.4 (except the URL DoS issue)</li>
</ul>
<p>Per our <a class="reference external" href="https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions">supported versions policy</a>, Django 1.5 and 1.6 are no longer receiving security updates.</p>
</div>
<div class="section" id="s-resolution">
<h3>Resolution</h3>
<p>Patches have been applied to Django's master development branch and to the 1.4, 1.7, and 1.8
release branches, which resolve the issues described above. The patches may be obtained
directly from the following changesets:</p>
<p>On the development master branch:</p>

<p>On the 1.8 release branch:</p>

<p>On the 1.7 release branch:</p>

<p>On the 1.4 release branch:</p>

<p>The following new releases have been issued:</p>

<p>Note: The first 1.7.9 wheel file that we uploaded was corrupt (it contained some files from 1.8). A corrected file was uploaded about 2 hours after the initial release.</p>
<p>The PGP key ID used for these releases is Tim Graham: 1E8ABDC773EDE252.</p>
</div>
<div class="section" id="s-general-notes-regarding-security-reporting">
<h3>General notes regarding security reporting</h3>
<p>As always, we ask that potential security issues be reported via
private email to <tt class="docutils literal">security@djangoproject.com</tt>, and not via Django's
Trac instance or the django-developers list. Please see <a class="reference external" href="https://www.djangoproject.com/security/">our security
policies</a> for further
information.</p>
</div>


        <a href="#top" class="backtotop"><i class="icon icon-chevron-up"/> Back to Top</a>
      </div>

      
  </div></body></html>